R/trackhub.R

```diff
@@ -346,26 +346,54 @@ TrackHubGenome <- function(trackhub, genome, create = FALSE, genomeRecord = Geno
 ###
 
 setClass("TrackContainer",
+         representation("SimpleList"),
+         prototype(elementType = "Track_OR_TrackContainer")
+)
+
+TrackContainer <- function() {
+    new("TrackContainer")
+}
+
+setMethod("[[", "TrackContainer", function(x, i, ...) {
+    getListElement(x, i, ...)
+})
+
+setReplaceMethod("[[", "TrackContainer", function(x, i, ..., value) {
+    setListElement(x, i, value)
+})
+
+setClass("Track",
          representation(
-                        trackList = "list",
-                        childrenList = "list",
-                        parentLookUp = "list"
-                        ),
-         prototype = list(
-                          trackList = list(),
-                          childrenList = list(),
-                          parentLookUp = list()
-         )
+                        track = "character",
+                        bigDataUrl = "character",
+                        children = "TrackContainer"
```

Comment:
I thought that Track objects were leaves in the tree and that only TrackContainer objects would contain children.